File: app/src/net/reichholf/dreamdroid/tv/BrowseItem.java
package net.reichholf.dreamdroid.tv;

import net.reichholf.dreamdroid.helpers.ExtendedHashMap;

import java.io.Serializable;

public class BrowseItem implements Serializable{
	public enum Type {
		Reload,
		Service,
		Movie,
		Profile,
		Preferences
	}

	public Type type;
	public ExtendedHashMap data;

	public BrowseItem(Type itemType, ExtendedHashMap itemData) {
		type = itemType;
		data = itemData;
	}
}

File: app/src/net/reichholf/dreamdroid/tv/BrowseItem.kt
package net.reichholf.dreamdroid.tv

/**
 * Settings kinds still used by the Compose TV hub. Typed service/movie payloads
 * from Phase 3.1a were unused after the hub rewrite.
 */
object BrowseItem {
    enum class Kind {
        Reload,
        Preferences,
        Profile,
    }
}
